test/es-module/test-esm-throw-undefined.mjs

```diff
@@ -0,0 +1,15 @@
+// Flags: --experimental-modules
+/* eslint-disable node-core/required-modules */
+import common from '../common/index.js';
+import assert from 'assert';
+
+async function doTest() {
+  try {
```

Comment:
@devsnek OK. BTW, I can't find any `async` and `await` in existing es-module tests. I don't know if they can be used.

Reply:
async/await is safe to use. you can test your tests with `python tools/test.py -J --mode=release path/to/your/test` e.g. `python tools/test.py -J --mode=release test/es-module/test-esm-throw-undefined.mjs`

Reply:
@devsnek I find in this case I can't use `assert.rejects`. It has the same problem as `assert.throws`, because when it comes to `undefined` or `null`, many results can be considered OK. See:

```js
assert.rejects(async () => {throw Error;}, null)
assert.rejects(async () => {throw Error;}, undefined)
assert.rejects(async () => {throw undefined;}, null)
assert.rejects(async () => {throw null;}, undefined)
```

All of these 4 are considered as "rejected". So it will make the test too loose.

Reply:
@devsnek Sorry just aware that I can use function `e => e === undefined` as the second parameter. Will modify test.